python/lsst/ip/isr/ampOffset.py

```diff
@@ -18,88 +18,406 @@
     )
-    ampEdgeWindow = pexConfig.Field(
-        doc="Pixel size of the sliding window used to generate rolling average amp offset values.",
+    ampEdgeWindow = Field(
+        doc="Pixel size of the sliding window used to generate rolling average amp offset values. It should "
+        "not exceed the shorter dimension of amps.",
         dtype=int,
-        default=512,
+        default=509,  # We need to reconsider this for every instrument (subaru, lsst, etc.)
     )
-    doBackground = pexConfig.Field(
+    doBackground = Field(
         doc="Estimate and subtract background prior to amp offset estimation?",
         dtype=bool,
         default=True,
     )
-    background = pexConfig.ConfigurableField(
+    background = ConfigurableField(
         doc="An initial background estimation step run prior to amp offset calculation.",
         target=SubtractBackgroundTask,
     )
-    doDetection = pexConfig.Field(
+    doDetection = Field(
         doc="Detect sources and update cloned exposure prior to amp offset estimation?",
         dtype=bool,
         default=True,
     )
-    detection = pexConfig.ConfigurableField(
+    detection = ConfigurableField(
         doc="Source detection to add temporary detection footprints prior to amp offset calculation.",
         target=SourceDetectionTask,
     )
 
 
-class AmpOffsetTask(pipeBase.Task):
-    """Calculate and apply amp offset corrections to an exposure.
-    """
+class AmpOffsetTask(Task):
+    """Calculate and apply amp offset corrections to an exposure."""
+
     ConfigClass = AmpOffsetConfig
     _DefaultName = "isrAmpOffset"
 
     def __init__(self, *args, **kwargs):
-        super().__init__(**kwargs)
-        # always load background subtask, even if doBackground=False;
-        # this allows for default plane bit masks to be defined
+        super().__init__(*args, **kwargs)
+        # Always load background subtask, even if doBackground=False;
+        # this allows for default plane bit masks to be defined.
         self.makeSubtask("background")
         if self.config.doDetection:
             self.makeSubtask("detection")
 
     def run(self, exposure):
         """Calculate amp offset values, determine corrective pedestals for each
-        amp, and update the input exposure in-place. This task is currently not
-        implemented, and should be retargeted by a camera specific version.
+        amp, and update the input exposure in-place.
+
+        Parameters
+        ----------
+        exposure: `lsst.afw.image.Exposure`
+            Exposure to be corrected for amp offsets.
+        """
+
+        # Generate an exposure clone to work on and establish the bit mask.
+        exp = exposure.clone()
+        bitMask = exp.mask.getPlaneBitMask(self.background.config.ignoredPixelMask)
+        self.log.info(
+            "Ignored mask planes for amp offset estimation: [%s].",
+            ", ".join(self.background.config.ignoredPixelMask),
+        )
```

Comment:
The `background` task will already print something like this in the logs if the log level is set to `DEBUG` mode:
```
DEBUG 2023-08-01T12:04:06.874-07:00 lsst.isr.ampOffset.background (isr:{instrument: 'HSC', detector: 42, exposure: 1228, ...})(subtractBackground.py:255) - Ignoring mask planes: BAD, SAT, INTRP, CR, EDGE, DETECTED, DETECTED_NEGATIVE, SUSPECT, NO_DATA
```
I'm not sure we want it repeated at INFO level too, do we?

Reply:
Probably not

Reply:
Removed